src/Microsoft.ML.AutoML/API/BinaryClassificationExperiment.cs

```diff
@@ -332,7 +338,7 @@ private SweepablePipeline CreateBinaryClassificationPipeline(IDataView trainData
 
     internal class BinaryClassificationRunner : ITrialRunner
```

Comment:
>BinaryClassificationRunner [](http://example.com/codeflow?start=19&length=26)

Should we use IDisposable with this class?

Reply:
That's a great question here. The IDisposable pattern is mainly because I want to make sure `MLContext.CancelExecuation` get called and set to null after the trial is finished while I don't want to explicitly call it's deconstructor or call GC. But I can go another route if you have any recommendation.

Reply:
Dispose pattern is not only for GC finalization or deconstruction. It is used to clean up any object when done using it. You already implemented the Dispose method. Having IDisposable will benefit with the `using ()` code pattern.